pjmedia/src/pjmedia/transport_ice.c

```diff
@@ -773,6 +1132,31 @@ static pj_status_t encode_session_in_sdp(struct transport_ice *tp_ice,
     	m->attr[m->attr_count++] = add_attr;
     }
 
+    /* Add trickle ICE attributes */
+    if (trickle) {
+	pj_ice_strans_state ice_state;
+	pj_bool_t end_of_cand;
+
+	/* Add media ID attribute "a=mid" */
+	attr = pjmedia_sdp_attr_find2(m->attr_count, m->attr, "mid", NULL);
+	if (!attr) {
+	    attr = pjmedia_sdp_attr_create(sdp_pool, "mid", &tp_ice->sdp_mid);
+	    pjmedia_sdp_attr_add(&m->attr_count, m->attr, attr);
+	}
+
+	ice_state = pj_ice_strans_get_state(tp_ice->ice_st);
+	end_of_cand = (ice_state == PJ_ICE_STRANS_STATE_READY ||
+		       ice_state == PJ_ICE_STRANS_STATE_SESS_READY);
+
+	status = pjmedia_ice_trickle_encode_sdp(sdp_pool, sdp_local,
+						&tp_ice->sdp_mid, NULL, NULL,
+						0, NULL, end_of_cand);
+	if (status != PJ_SUCCESS) {
+	    pj_assert(!"pjmedia_ice_trickle_encode_sdp() failed");
```

Comment:
Not sure why we need to assert here.
There are already assertions in place inside the function itself, which is the more appropriate place, and there's also a legitimate error as well (if too many media) although it's a rare case.

Reply:
Replaced the assertion with an error log.